src/java/org/apache/cassandra/net/FrameDecoder.java

```diff
@@ -190,6 +190,11 @@ boolean isConsumed()
     abstract void decode(Collection<Frame> into, ShareableBytes bytes);
     abstract void addLastTo(ChannelPipeline pipeline);
 
+    public boolean isActive()
```

Comment:
Consider a javadoc comment here articulating when we expect the Frame Decoder to be active vs. non. We didn't have that prior but we also didn't expose it as a public API and use it this heavily.

Reply:
Agreed. Some brief description of what being "active" entails would be helpful.